ci/e2e-tests.sh

```diff
@@ -0,0 +1,615 @@
+        exit 1
+    fi
+    echo "Artifacts URL: $artifacts_url" >&2
+
+    case "$OS" in
+        'centos:7')
+            artifact_name='centos-7'
+            ;;
+
+        'debian:9')
+            artifact_name='debian-9-slim'
+            ;;
+
+        'debian:10')
+            artifact_name='debian-10-slim'
+            ;;
+
+        'ubuntu:18.04')
+            artifact_name='ubuntu-18.04'
+            ;;
+
+        'ubuntu:20.04')
+            artifact_name='ubuntu-20.04'
+            ;;
+
+        *)
+            echo "Unsupported OS $OS" >&2
+            exit 1
+    esac
+    artifact_name="packages_${artifact_name}_amd64"
+
+    echo 'Getting artifact download URL...' >&2
+    artifact_download_url="$(
+        github_curl -L \
+            -H 'accept: application/vnd.github.v3+json' \
+            "$artifacts_url" |
+            jq \
+                --arg artifact_name "$artifact_name" \
+                -r \
+                '.artifacts[] | select(.name == $artifact_name) | .archive_download_url'
+    )"
+    if [ -z "$artifact_download_url" ]; then
+        echo "Could not find artifact for OS $OS" >&2
+        exit 1
+    fi
+    echo "Artifact download URL: $artifact_download_url" >&2
+
+    echo 'Downloading artifact...' >&2
+    github_curl -L \
+        -o package.zip \
+        "$artifact_download_url"
+    echo 'Downloaded artifact' >&2
+
+
+    echo 'Extracting package...' >&2
+    case "$OS" in
+        'centos:7')
+            unzip -j package.zip 'centos7/amd64/aziot-identity-service-*.x86_64.rpm' -x '*-debuginfo-*.rpm' '*-devel-*.rpm' >&2
+            printf '%s/%s\n' "$PWD" aziot-identity-service-*.x86_64.rpm
+            ;;
+
+        'debian:9')
+            unzip -j package.zip 'debian9/amd64/aziot-identity-service_*_amd64.deb' >&2
+            printf '%s/%s\n' "$PWD" aziot-identity-service_*_amd64.deb
+            ;;
+
+        'debian:10')
+            unzip -j package.zip 'debian10/amd64/aziot-identity-service_*_amd64.deb' >&2
+            printf '%s/%s\n' "$PWD" aziot-identity-service_*_amd64.deb
+            ;;
+
+        'ubuntu:18.04')
+            unzip -j package.zip 'ubuntu1804/amd64/aziot-identity-service_*_amd64.deb' >&2
+            printf '%s/%s\n' "$PWD" aziot-identity-service_*_amd64.deb
+            ;;
+
+        'ubuntu:20.04')
+            unzip -j package.zip 'ubuntu2004/amd64/aziot-identity-service_*_amd64.deb' >&2
+            printf '%s/%s\n' "$PWD" aziot-identity-service_*_amd64.deb
+            ;;
+
+        *)
+            echo "Unsupported OS $OS" >&2
+            exit 1
+    esac
+    echo 'Extracted package' >&2
+}
+
+
+test_name="$1"
+
+
+run_id="$OS:$GITHUB_RUN_ID:$GITHUB_RUN_NUMBER:$test_name"
+echo "Run ID: $run_id" >&2
+
+
+# Temp directory used as scratch space to store the downloaded package from GitHub
+# and the config files for the package that are scp'd to the test VM.
+rm -rf /tmp/iot-identity-service-e2e-tests
```

Comment:
How about `mktemp -d`?

Reply:
It's convenient to have a fixed directory for debugging.

Thinking about it now, it would be nice to be able to run multiple runs at the same time. I'll add the `$run_id` to the directory name.

Reply:
Done.